### PR TITLE
Simpler TreeTrie.isEmpty() method

### PR DESCRIPTION
**Pull request by joakime**
Avoid creating (Key)Set to just test for empty.

### CODE REVIEW

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TreeTrie.java [hunk 1]
```diff
@@ -276,7 +276,31 @@ private V getBest(Node<V> node, byte[] b, int offset, int len)
...
+                        return true;
+                }
+            }
+            for (int i = t._nextOther.size(); i-- > 0; )
+            {
+                if (hasAnyKey(t._nextOther.get(i)))
+                    return true;
+            }
+        }
+        return false;
```
Comment (gregw) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TreeTrie.java]: probably invert the boolean and rename isEmpty(Node)

**Review by gregw — APPROVED** [review]
Other than a rename LGTM

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TreeTrie.java (hunk 1 above)
Comment (joakime) [reply jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TreeTrie.java]: Inverted / Renamed

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TreeTrie.java [hunk 2]
```diff
@@ -276,7 +276,31 @@ private V getBest(Node<V> node, byte[] b, int offset, int len)
...
+    }
+
+    private boolean isEmpty(Node<V> t)
+    {
+        if (t != null)
+        {
+            if (t._key != null)
+                return false;
+
+            for (int i = 0; i < INDEX; i++)
```
Comment (lorban) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TreeTrie.java:289]: Is there some test coverage for this logic?

**Review by lorban — APPROVED** [review]
This looks alright, but I think we should add some testing to cover the two loops in `isEmpty(Node)`.

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TreeTrie.java (hunk 2 above)
Comment (joakime) [reply jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TreeTrie.java:289]: Yup, the first loop is 100% covered by the 2 new test cases.
The second loop executes, but nothing can trigger the internal isEmpty() logic with the test cases as it stands.

**Comment by gregw** [comment]
I just pushed 3d4bb492d0c7ffff350ad627fbf3a33382c999d3 which slightly improves the test coverage

**Pull request merged**